inst/doc/shortcuts.Rmd

```diff
@@ -0,0 +1,107 @@
+---
+title: "`ggplot2` Shortcuts"
+author: "Jonathan Carroll"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{`ggplot2` Shortcuts}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+---
+
+This package allows easy access to some common `ggplot2` tasks. For example, rotating the `x` axis labels is a very frequently looked up task, and we can make it easier. If we create a simple `ggplot2` plot
+
+```{r, message = FALSE}
+library(ggplot2)
+library(cowplot)
+library(labelled)
+library(ggeasy)
+
+p <- ggplot(mtcars, aes(hp, mpg)) + geom_point()
+```
+
+then by default, this looks like
+
+```{r, fig.width = 6, fig.height = 6}
+p + labs(title = "ggplot2 default")
+```
+
+We can perform various rotations though
+
+```{r, fig.width = 8, fig.height = 8}
+p1 <- p + 
+    easy_rotate_x_labels() + 
+    labs(title = "default rotation")
+p2 <- p + 
+    easy_rotate_x_labels(angle = 45, side = "right") + 
+    labs(title = "angle = 45")
+p3 <- p + 
+    easy_rotate_x_labels("startattop") + 
+    labs(title = "text starts at top")
+p4 <- p + 
+    easy_rotate_x_labels("startatbottom") + 
+    labs(title = "text starts at bottom")
+    
+plot_grid(p1, p2, p3, p4, nrow = 2)
+```
+
+## Removing legends
+
+Removing legends is made easier by the `easy_remove_legend` function.  When called without arguments, all legends are removed (equivalent to `theme(legend.position = "none")`). Alternatively, the names of aesthetics for which legends should be removed can be passed.
+
+```{r, fig.width = 8, fig.height = 8}
+p <- ggplot(mtcars, aes(wt, mpg, colour = cyl, size = hp)) +
+    geom_point()
+
+p1 <- p + 
+    labs(title = "With all legends")
+p2 <- p + 
+    easy_remove_legend() + 
+    labs(title = "Remove all legends")
+p3 <- p + 
+    easy_remove_legend(size) + 
+    labs(title = "Remove size legend")
+p4 <- p + 
+    easy_remove_legend(size, color) + 
+    labs(title = "Remove both legends specifically")
+
+plot_grid(p1, p2, p3, p4, nrow = 2)
+```
+
+## Plot Labels
+
+```{r, fig.width = 8, fig.height = 8}
+iris_labs <- iris
+
+lbl <- c('Sepal Length','Sepal Width','Petal Length','Petal Width','Flower Species')
+
+var_label(iris_labs) <- split(lbl,names(iris_labs))
+
+p <- 
+     ggplot(iris_labs,aes(x=Sepal.Length,y=Sepal.Width))+
+     geom_line(aes(colour=Species)) +
+     theme(text=element_text(size=7))
```

Comment:
`size=7` makes these small for me. In general I'd like to avoid adding any commands which don't highlight the difference between the 'easy' way and the 'hard' way, even if they look nicer.